elastic4s-effect-cats-2/src/main/scala/com/sksamuel/elastic4s/cats/effect/instances/CatsEffectInstances.scala

```diff
@@ -12,6 +12,9 @@ trait CatsEffectInstances {
 
   implicit def catsEffectExecutor[F[_]: Async]: Executor[F] =
     new CatsEffectExecutor[F]
+
+  //this needs to be at the bottom
```

Comment:
I don't know why the ordering matters.

An alternative implementation with a lower priority trait `CatsEffectInstances0` also works:
```scala
trait CatsEffectInstances extends CatsEffectInstances0 {
  implicit def catsFunctor[F[_]: CatsFunctor]: Functor[F] = new Functor[F] {
    override def map[A, B](fa: F[A])(f: A => B): F[B] = CatsFunctor[F].map(fa)(f)
  }

  implicit def catsEffectExecutor[F[_]: Async]: Executor[F] =
    new CatsEffectExecutor[F]
}

trait CatsEffectInstances0 {
  implicit val ioExecutor: Executor[IO] = new CatsEffectExecutor[IO]
}
```
but I'm equally confused as to why the more specific `ioExecutor` needs to be in the lower and not the higher priority trait for it to work.

Reply:
I  want to do https://github.com/sksamuel/elastic4s/issues/2379 as well for 8.0. Maybe that's easier to do now ?

Reply:
> I want to do #2379 as well for 8.0. Maybe that's easier to do now ?

I think they are mutually exclusive. For #2379 most of the work seem to revolve around refactoring `HttpClient`.

The main pain points with the current `execute` for me are:
- Defaulting to `Future`
The default `Future` typeclass instances in the companion objects for `Functor`/`Executor` seems conflict prone to me in terms of implicit resolution. Maybe you could drop the default instances and have 2 execution methods instead (`execute`/`executeF`) where one always return a `Future`. Or just keep to 1 and have users manually import `Future` instances.
2. The type parameters `[T, U, F[_]]`
Having to be explicit with the effect type `F` is fine but it forces you to have to figure out what `T` and `U` are. Being able to explicit on `F` without `T` and `U` would be nice, perhaps something in the direction of `client.effect[IO].execute(...)`